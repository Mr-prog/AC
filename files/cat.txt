  section_program
  start:
    READ_PORT
    WRITE_PORT R4
    NOP