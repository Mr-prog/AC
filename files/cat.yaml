; cat.asm
; Считать данные с порта ввода и вывести их на экран

start:
    READ_PORT PIN, R1          ; Чтение данных из порта ввода
    JZ R1, end                 ; Если данные закончились (нулевое значение), завершить
    WRITE_PORT R1, POUT        ; Вывод данных на экран
    JUMP start                 ; Продолжить чтение данных

end:
    NOP                        ; Завершение программы


section_program
READ_PORT

MOV R4 R3
WRITE_PORT
NOP
section_data