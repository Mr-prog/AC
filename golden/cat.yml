in_source: |
  section_program
  start:
    READ_PORT
    WRITE_PORT R4
    NOP
in_stdin: '"C","a","t"'
out_code: |
  Instruction counter: 0, READ_PORT []
  Instruction counter: 1, WRITE_PORT ['R4']
  Instruction counter: 2, NOP []
out_stdout: 'Cat'
out_log: |
  DEBUG    root:machine.py:234 READ_PORT [] {PC: 0, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: , REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:234 WRITE_PORT ['R4'] {PC: 1, REG0: 0, REG1: , REG2: , REG3: , REG4: Cat, REG5: , REG6: , REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:234 NOP [] {PC: 2, REG0: 0, REG1: , REG2: , REG3: , REG4: Cat, REG5: , REG6: , REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['C', 'a', 't']}
